src/test/ui/inference/issue-83606.stderr

```diff
@@ -5,6 +5,11 @@ LL |     let _ = foo("foo"); //<- Do not suggest `foo::<N>("foo");`!
    |         -   ^^^ cannot infer the value of const parameter `N` declared on the function `foo`
    |         |
    |         consider giving this pattern the explicit type `[_; N]`, where the const parameter `N` is specified
+   |
+help: consider specifying the const argument
+   |
+LL |     let _ = foo::<N>("foo"); //<- Do not suggest `foo::<N>("foo");`!
```

Comment:
It seems like this PR re-introduces #83606.

Reply:
It doesn't, 83606 is no longer an issue because we can now specify const generics even if an `impl Trait` argument type exists. I have removed the comment which no longer applies.

It might be better to just remove this test, but I figured it is good to test that we don't ICE or something.